packages/grid/src/Grid.test.tsx

```diff
@@ -0,0 +1,223 @@
+      },
+      {
+        prop: 'colEnd',
+        style: 'gridColumn',
+        values: [1, 2, 3],
+        expectedValues: ['auto / 1', 'auto / 2', 'auto / 3'],
+      },
+      {
+        prop: 'rowStart',
+        style: 'gridRow',
+        values: [1, 2, 3],
+        expectedValues: ['1 / auto', '2 / auto', '3 / auto'],
+      },
+      {
+        prop: 'rowEnd',
+        style: 'gridRow',
+        values: [1, 2, 3],
+        expectedValues: ['auto / 1', 'auto / 2', 'auto / 3'],
+      },
+    ];
+
+    for (const { prop, style, values, expectedValues } of gridItemProperties) {
+      describe(prop, () => {
+        it.each(
+          values.map((value, index) => ({
+            [prop]: value,
+            expectedValue: expectedValues?.[index] ?? value,
+          })),
+        )(
+          `should apply ${style} when ${prop} prop is $${prop}`,
+          ({ [prop]: value, expectedValue }) => {
+            render(
+              <Grid.Root>
+                <Grid.Item data-testid="grid-item" {...{ [prop]: value }}>
+                  item 1
+                </Grid.Item>
+              </Grid.Root>,
+            );
+
+            const gridItem = screen.getByTestId('grid-item');
+            expect(gridItem).toHaveStyle({ [style]: expectedValue });
+          },
+        );
+      });
+    }
+  });
+
+  describe('style overrides', () => {
+    it('should allow style prop to override default styles', () => {
+      const customStyle = {
+        backgroundColor: 'red',
+        padding: '20px',
+      };
+
+      render(
+        <Grid.Root data-testid="grid-container" style={customStyle}>
+          <Grid.Item>item 1</Grid.Item>
+        </Grid.Root>,
+      );
+
+      const gridContainer = screen.getByTestId('grid-container');
+      expect(gridContainer).toHaveStyle(customStyle);
+    });
+  });
+
+  describe('nested Grid components', () => {
+    it('should render nested Grid components correctly', () => {
+      render(
+        <Grid.Root data-testid="parent-grid" templateColumns="1fr 1fr">
+          <Grid.Item>
+            <Grid.Root data-testid="child-grid" templateRows="1fr 1fr">
+              <Grid.Item>nested item 1</Grid.Item>
+              <Grid.Item>nested item 2</Grid.Item>
+            </Grid.Root>
+          </Grid.Item>
+        </Grid.Root>,
+      );
+
+      const parentGrid = screen.getByTestId('parent-grid');
+      const childGrid = screen.getByTestId('child-grid');
+
+      expect(parentGrid).toHaveStyle({ '--grid-template-columns': '1fr 1fr' });
+      expect(childGrid).toHaveStyle({ '--grid-template-rows': '1fr 1fr' });
+    });
+  });
+
+  describe('polymorphic behavior', () => {
+    it.each(['span', 'nav', 'button', 'input', 'label', 'div'])(
+      'should render as %s element when asChild is true',
+      (element) => {
+        render(
+          <Grid.Root data-testid="grid-container" asChild>
+            {createElement(element)}
+          </Grid.Root>,
+        );
```

Comment:
```suggestion
      (Element) => {
        render(
          <Grid.Root data-testid="grid-container" asChild>
            <Element />
          </Grid.Root>,
        );
```
이렇게 하면 createElement 없어도 되지 않을까?

Reply:
어 맞네 ㅋㅋㅋㅋ 왜 이렇게 했지

Reply:
tag name만으로도 Element 선언을 할 수 있다는 것을 처음 알게됨!!